test/hotspot/jtreg/runtime/Thread/TestAlwaysPreTouchStacks.java

```diff
@@ -103,83 +163,30 @@ public static void main(String[] args) throws Exception {
-                Matcher m = pat.matcher(line);
-                if (m.matches()) {
-                    long reserved = Long.parseLong(m.group(1));
-                    long committed = Long.parseLong(m.group(2));
-                    System.out.println(">>>>> " + line + ": " + reserved + " - " + committed);
-                    // This is a bit fuzzy: even with PreTouch we don't commit the full range of what NMT counts
-                    // as thread stack. But without pre-touching, the thread stacks would be committed to about 1/5th
-                    // of their reserved size. Requiring them to be committed for over 3/4th shows that pretouch is
-                    // really working.
-                    if (preTouch && (double)committed < ((double)reserved * 0.75)) {
-                        throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
-                    } else if (!preTouch && (double)committed > ((double)reserved * 0.50)){
-                        throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
-                    }
-                    // Added sanity tests: we expect our test threads to be still alive when NMT prints its final
-                    // report, so their stacks should dominate the NMT-reported total stack size.
-                    long max_reserved = memoryCeilingMB * 3 * MB;
-                    long min_reserved = memoryCeilingMB * MB;
-                    if (reserved >= max_reserved || reserved < min_reserved) {
-                        throw new RuntimeException("Total reserved stack sizes outside of our expectations (" + reserved +
-                                                   ", expected " + min_reserved + ".." + max_reserved + ")");
-                    }
-                    foundLine = true;
-                    break;
-                }
-            }
-            if (!foundLine) {
-                throw new RuntimeException("Did not find expected NMT output");
-            }
-        }
-
+          ReservedCommitted pretouch_result = runPreTouchTest(true);
+          ReservedCommitted no_pretouch_result = runPreTouchTest(false);
+          if (pretouch_result.reserved == 0 || no_pretouch_result.reserved == 0) {
+            throw new RuntimeException("Could not run with PreTouch flag.");
+          }
+          double ratio_with = ((double)pretouch_result.committed) / pretouch_result.reserved;
+          double ratio_without = ((double)no_pretouch_result.committed) / no_pretouch_result.reserved;
+          System.out.println("ratio with PreTouch: " + ratio_with + " w/out: " + ratio_without);
+          if (ratio_without > 0.50) {
+            throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
+          }
+          if (ratio_with < ratio_without) {
+            throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
+          }
```

Comment:
If we don't save the Reserved and just look at it for each run independently, we get back to the original test which has problems in macos-aarch64 environments. We decided/agreed to have two tests and save the amounts and then compare them to check if pretouch was done or not.
Maybe I misunderstood you.

Reply:
@afshin-zafari I just mean that since you compare the committed values of two runs, one with and one without pretouch, you don't really need to compare the reserved values or calculate the reserved to committed ratio. In fact, in both cases, reserved should be near identical. 

All you really need to do is to make sure that committed (with pretouch) is > (committed (without pretouch) + expected delta). So, committed with pretouch should be at least 80MB larger than committed without pretouch.

Am I making a thinking error here?

Reply:
Sorry for confusion. It is clear to me now. The changes are pushed.